spec/app/controllers/courses_controller/controller_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+RSpec.describe CoursesController, type: :request do
+  let(:request_payload) { 
+    {
+      course_uuid: given_course_data.uuid,
+      ecosystem_uuid: given_course_data.ecosystem_uuid
+    }
+  }
+
+  let(:given_course_data) {
+    build(:course, ecosystem_uuid: ecosystem.uuid)
+  }
+
+  let(:ecosystem) {
+    build(:ecosystem)
+  }
```

Comment:
Using `build` to construct what amounts to two uuids seems excessive; I've prefer the more basic:
```ruby
  let(:request_payload) {
    {
      course_uuid:    given_course_uuid,
      ecosystem_uuid: given_ecosystem_uuid,
    }
  }

  let(:given_course_uuid)    { SecureRandom.uuid.to_s }
  let(:given_ecosystem_uuid) { SecureRandom.uuid.to_s }
```